scribejava-core/src/main/java/com/github/scribejava/core/model/HttpClient.java

```diff
@@ -11,6 +11,9 @@
                                String bodyContents, OAuthAsyncRequestCallback<T> callback,
                                OAuthRequestAsync.ResponseConverter<T> converter);
 
+    <T> Future<T> executeAsync(String userAgent, Map<String, String> headers, Verb httpVerb, String completeUrl,
+                               byte[] bodyContents, OAuthAsyncRequestCallback<T> callback,
+                               OAuthRequestAsync.ResponseConverter<T> converter);
```

Comment:
and whats for java.io.File?
I have refactored ScribeJava a bit (uncouple OAuthRequest and Service ) https://github.com/scribejava/scribejava/commit/36c33b6001fd34a0fedc3e4df901e352aa780949

And add your changes with java.io.Files support as well https://github.com/scribejava/scribejava/commit/b7d0c6dd510c23457fa8ad9da46a6748749aa032